src/Microsoft.Azure.WebJobs.Host/Bindings/FunctionBindingContext.cs

```diff
@@ -48,5 +66,39 @@ public class FunctionBindingContext
         /// The short name of the current function. 
         /// </summary>
         public string MethodName { get; private set; }
+
+        /// <summary>
+        /// The service provider for the current function invocation scope.
+        /// </summary>
+        public IServiceProvider InstanceServices { get; set; }
+
+        /// <summary>
+        /// Creates an object instance with constructor arguments provided in the
+        /// method call and from the current function context scoped services.
+        /// </summary>
+        /// <param name="type">The type to instantiate.</param>
+        /// <param name="parameters">Parameters not provided by the function service provider.</param>
+        /// <returns>An instance of the type specified.</returns>
+        public object CreateObjectInstance(Type type, params object[] parameters)
+        {
+            return ActivatorUtilities.CreateInstance(_functionInvocationServices, type, parameters);
```

Comment:
When is CreateObjectInstance called? Also, can you clarify what do you mean by 'Parameters not provided by the function service provider.'

Reply:
This is a new public API exposed to extension authors (and other components in the pipeline) to instantiate objects, which will leverage DI. This helps them avoid direct instantiation/`Activator` calls while enabling automatic service injection.

`Parameters not provided by the function service provider` are all parameters directly provided by the invoker, instead of the DI service provider.